src/fmt.zig

```diff
@@ -270,6 +270,7 @@ const Formatter = struct {
         for (statement_slice) |s| {
             const region = fmt.nodeRegion(@intFromEnum(s));
             _ = try fmt.flushCommentsBefore(region.start);
+            try fmt.ensureNewline();
```

Comment:
Is this saying we must always have a newline between statements? I'm not sure that is the plan. Maybe we should discuss the intended design on Zulip?

Reply:
For the correct snapshots I haven't noticed any regression.
In practice, what this does is fixing code that is originally written on the same line.
It also adds a newline for malformed statements, which causes the snapshot diffs below, but they were already wrong before.
For example:

```
module[]0 f
```
becomes
```
module []
0
f
```
instead of just concatenating everything into a single line as before
```
module []0f
```